test/parallel/test-assert-match.js

```diff
@@ -0,0 +1,105 @@
+// Currently in sync with Node.js test/parallel/test-assert.js
+// https://github.com/nodejs/node/commit/2a871df3dfb8ea663ef5e1f8f62701ec51384ecb
+
+'use strict';
+
+require('../common');
+const assert = require('../../assert');
+
+// Multiple assert.match() tests.
+{
+  assert.throws(
+    () => assert.match(/abc/, 'string'),
+    {
+      code: 'ERR_INVALID_ARG_TYPE',
+      message: 'The "regexp" argument must be of type RegExp. Received type string'
```

Comment:
There is a slight difference in the message generated by this implementation and the one in Node.js. It should be actually be:
`The "regexp" argument must be an instance of RegExp. Received type string ('string')`
I think this should be updated in a separate PR as it touches the `ERR_INVALID_ARG_TYPE` error type which is not super-related to `assert.match()`